scripts/isolate_tests.py

```diff
@@ -106,13 +84,18 @@ def extract_and_write(f, path):
     write_cases(f, cases)
 
 if __name__ == '__main__':
+    if len(sys.argv) == 1:
+        print("Usage: " + sys.argv[0] + " path-to-file-or-folder-to-extract-code-from [docs]")
+        exit(1)
+
     path = sys.argv[1]
     docs = False
     if len(sys.argv) > 2 and sys.argv[2] == 'docs':
         docs = True
```

Comment:
If you want to get sane argument parsing with little effort, please use `argparse`.

This basically does the same as the above but with better checks and can print `--help`:

```python
from argparse import ArgumentParser

script_description = (
    "Reads C++ or RST source files and writes all multi-line strings into individual files. "
    "This can be used to extract the Solidity test cases into files for e.g. fuzz testing."
)

parser = ArgumentParser(description=script_description)
parser.add_argument(dest='path', help="Path to file or directory to look for code in.")
parser.add_argument(dest='docs', nargs='?')
options = parser.parse_args()

path = options.path
docs = (options.docs == 'docs')
```

I'm already using it this way in https://github.com/ethereum/solidity/blob/develop/scripts/bytecodecompare/prepare_report.py (though it's more complex there because there are more parameters).

Reply:
Also, this weird `docs` parameter should really be replaced with a proper flag (though this changes the interface of the script):
```python
parser.add_argument('--docs', dest='docs', default=False, action='store_true', help="Assume that all processed files are ReStructuredText documents and try to extract code snippets from them.")
```
Or, maybe like this:
```python
from enum import Enum

class ExtractionTarget(Enum):
    DOCS = 'docs'
    SOLIDITY = 'solidity'

...

parser.add_argument(
    '--target',
    dest='extraction_target',
    default=ExtractionTarget.SOLIDITY.value,
    choices=[t.value for t in ExtractionTarget],
    help="Targets to extract code from. docs = code snippets in ReStructuredText documents, solidity = .sol files"
)
```